vmdb/app/models/mixins/ems_openstack_mixin.rb

```diff
@@ -78,6 +78,14 @@ def verify_api_credentials(options={})
     rescue Excon::Errors::Unauthorized => err
       $log.error("MIQ(#{self.class.name}.verify_api_credentials) Error Class=#{err.class.name}, Message=#{err.message}")
       raise MiqException::MiqEVMLoginError, "Login failed due to a bad username or password."
```

Comment:
Yes, I agree. Until/if we delineate missing required part of credentials and login failures due to bad/user/password.